src/pages/ServiceDetails/ServiceMetrics.tsx

```diff
@@ -226,9 +243,14 @@ class ServiceMetrics extends React.Component<ServiceId, ServiceMetricsState> {
   dismissAlert = () => this.setState({ alertDetails: undefined });
 
   render() {
+    const { store } = this.context;
+    store.subscribe(() => {
```

Comment:
Does it really has to be subscribed in `render`?
`render` can be called many times in the component's lifecycle. I guess subscription has to be called only once. So, constructor or `componentDidMount` is a better fit, no?

Reply:
I don't know ^^ I am new in redux, This is one of my questions to @mtho11 and @vnugent how to handle with the subscribe(), I don't like to have them in the render, but They should be in each component that we wanna update if the store change

Reply:
It doesn't have to be subscribed to in render. A connected component will automatically re-render itself for the `mapStateToProps` state fields.